File: magmac/src/java/magma/api/collect/stream/EmptyHead.java
package magma.api.collect.stream;

import magma.api.option.Option;
import magma.lang.Instantiator;

import static magma.api.option.None.None;

public class EmptyHead<T> implements Head<T> {
    @Instantiator
    public static <T> Head<T> EmptyHead() {
        return new EmptyHead<>();
    }

    @Override
    public Option<T> head() {
        return None();
    }
}

File: src/node/magma/api/collect/head/EmptyHead.ts
/*[
	Actual, 
	Characters, 
	Collector, 
	Console, 
	EmptyHead, 
	Files, 
	Lists, 
	Namespace, 
	Strings
]*/
import { Head } from "../../../../magma/api/collect/head/Head";
import { Option } from "../../../../magma/api/option/Option";
import { None } from "../../../../magma/api/option/None";
export class EmptyHead<T> implements Head<T> {
	next(): Option<T> {
		return new None<T>(/*auto*/);
	}
}
